src/pages/pos-users/forms/PosUsersForm.js

```diff
@@ -34,32 +36,35 @@ export default function PosUsersForm({ labels, maxAccess, userId, window }) {
     enableReinitialize: true,
     validateOnChange: true,
     validationSchema: yup.object({
-      userId: yup.string().required(' '),
-      posId: yup.string().required(' ')
+      userId: yup.string().required(),
+      posId: yup.string().required()
     }),
     onSubmit: async obj => {
-      const userId = obj.userId
-
-      const response = await postRequest({
-        extension: PointofSaleRepository.PosUsers.set,
-        record: JSON.stringify(obj)
-      })
-      toast.success('Record Saved Successfully')
-
-      window.close()
-
-      invalidate()
+      try {
+        await postRequest({
+          extension: PointofSaleRepository.PosUsers.set,
+          record: JSON.stringify(obj)
+        })
+        toast.success(platformLabels.Saved)
+  
+        window.close()
+  
+        invalidate()
+      } catch (error) {}
     }
   })
+
+  const editMode = !!formik.values.recordId && !!recordId;
+
   useEffect(() => {
     ;(async function () {
       try {
-        if (userId) {
+        if (record && record.userId && recordId) {
           const res = await getRequest({
             extension: PointofSaleRepository.PosUsers.get,
-            parameters: `_userId=${userId}`
+            parameters: `_userId=${record.userId}`
           })
-          formik.setValues({ ...res.record, recordId: res?.record?.userId })
+          formik.setValues({ ...res.record, recordId: res.record.userId })
```

Comment:
use setFieldvalue instead setValues

Reply:
here we are setting the res.record + the recordId